Make icons in ProfileLink clickable

MehekFatima opened the pull request:
- Added `target="_blank"` and `rel="noopener noreferrer"` to `<a>` properties:This ensures that external links open in a new tab and helps avoid security risks by preventing the new page from gaining access to the referring page's context.

- Refactored CSS for cleaner structure:Removed redundant display: flex and `align-items: center` from `.profile__header__link` as the alignment is now handled within the `<a>` tag. 
- Made icons in ProfileLink clickable: Icons are now wrapped within the `<a>` tag, making them clickable along with the text.

alcpereira commented on src/components/Profile/ProfileLink/ProfileLink.css:
```diff
@@ -12,4 +12,5 @@
 .profile__header__link a {
   text-decoration: none;
   color: inherit;
+  display: contents;
```
This solution seems a bit hacky, and the gap between the icon and the text is not clickable.

alcpereira commented on src/components/Profile/ProfileLink/ProfileLink.css:
What do you think about restructuring this file? For example the `a` is not using CSS nesting while the `span` is.

alcpereira commented on src/components/Profile/ProfileLink/ProfileLink.tsx:
```diff
@@ -29,8 +29,10 @@ const ProfileLink = ({ type, link }: ProfileLinkProps) => {
     <div className="profile__header__link">
-      {icon}
-      <a href={`https://${link}`}>{type}</a>
+      <a href={`https://${link}`} target="_blank" rel="noopener noreferrer">
```
Nit pick: Please mention and explain this type of changes in the PR. While this is more than welcome, it deserves a mention for reviewers and for posterity.

MehekFatima commented:
I have made the following changes based on your feedback:

- Refactored `ProfileLink` Component: Changed the implementation to use a single `<a>` tag wrapping both the icon and text, addressing the issue with `display: contents`.
- Reorganized the CSS for better structure and consistency. The `<a>` tag now uses flexbox for alignment, and spacing between the icon and text is adjusted using margin.

alcpereira commented on src/components/Profile/ProfileLink/ProfileLink.css:
```diff
@@ -3,13 +3,16 @@
+
+.profile__header__link a span {
+  font-size: 14px;
+  margin-left: 0.5em; 
```
We are using `gap` in the project, margin is usually not recommended in modern CSS ([see this](https://youtu.be/KVQMoEFUee8?si=kVjm_vw8YK0jOeXM)).

alcpereira commented on src/components/Profile/ProfileLink/ProfileLink.css:
```diff
@@ -3,13 +3,16 @@
   color: inherit;
 }
+
+.profile__header__link a span {
```
The suggested change to edit the CSS and re-use CSS nesting was to do something along those lines:
```css
.profile__header__link {
  ...

  a {
    ...
  }

  span {
    ...
  }
}
```

alcpereira commented on src/components/Profile/ProfileLink/ProfileLink.css:
```diff
@@ -3,13 +3,16 @@
   align-items: center;
   gap: 0.5em;
```
Note that this gap value is for the spacing between the icon and the text, you might want to reuse this.

alcpereira commented:
Feel free to contact me on Discord ([project thread link](https://discord.com/channels/735923219315425401/1214333538594328676)) if you have any questions or need help.

MehekFatima commented:
I have updated the CSS based on the feedback:

-  The spacing between the icon and text is now managed using gap instead of margin.

-  The CSS has been restructured to use nesting for better organization and consistency.

Please review the updated styles and let me know if there are any further adjustments needed.

Thank you!

MehekFatima commented:
Hi @alcpereira ,

I've applied the requested changes, including removing the redundant gap and refactoring the CSS to use nesting. Could you please review the updates when you get a chance?

Thanks!

alcpereira commented:
@MehekFatima thanks for applying the changes.
Only 2 things left before merging it:
- Edit the PR description
  - Explain the changes regarding the `a` properties (`target` and `rel`)
  - Remove the outdated changes (regarding the `display: contents` for example), this information is already in the PR comments history
- Squash your commits into a single commit respecting Conventional commits, I would suggest for example `feat: make icons in profilelink clickable` 

For future PRs, I suggest you use a branch in your fork, currently you are using `main` which will cause issues if you work on more than on PR at the time or if some changes are merged before yours.

MehekFatima commented:
@alcpereira,

I’ve updated the PR description and addressed the changes for `<a>` properties. The commits have been squashed into a single commit with the message `feat: make icons in ProfileLink clickable`. 

I’ll use branches for future PRs to avoid conflicts.

Thanks!

alcpereira commented:
Thanks for the contribution 🙏
